app/components/Authentication/AwsEc2/AwsEc2.jsx

```diff
@@ -1,27 +1,450 @@
+        endpoint: undefined,
+        secret_key: ''
+    };
+
+    roleConfigSchema = {
+        role: '',
+        bound_ami_id: undefined,
+        bound_account_id: undefined,
+        bound_region: undefined,
+        bound_vpc_id: undefined,
+        bound_subnet_id: undefined,
+        bound_iam_role_arn: undefined,
+        bound_iam_instance_profile_arn: undefined,
+        role_tag: undefined,
+        ttl: undefined,
+        period: undefined,
+        policies: [],
+        allow_instance_migration: undefined,
+        disallow_reauthentication: undefined
+    }
+
     constructor(props) {
         super(props);
         this.state = {
-
+            baseUrl: `/auth/aws-ec2/${this.props.params.namespace}/`,
+            baseVaultPath: `auth/${this.props.params.namespace}`,
+            ec2Roles: [],
+            configObj: this.ec2ConfigSchema,
+            newConfigObj: this.ec2ConfigSchema,
+            newRoleConfig: this.roleConfigSchema,
+            newSecretBtnDisabled: false,
+            openNewRoleDialog: false,
+            openEditRoleDialog: false,
+            deleteUserPath: ''
         };
 
+        _.bindAll(
+            this,
+            'listEc2Roles',
+            'getEc2AuthConfig',
+            'createUpdateConfig',
+            'createUpdateRole'
+        );
+
+    }
+
+    listEc2Roles() {
+        tokenHasCapabilities(['list'], `${this.state.baseVaultPath}/role`)
+            .then(() => {
+                callVaultApi('get', `${this.state.baseVaultPath}/role`, { list: true }, null)
+                    .then((resp) => {
+                        let roles = resp.data.data.keys;
+                        this.setState({ ec2Roles: _.valuesIn(roles) });
+                    })
+                    .catch(snackBarMessage);
+            })
+            .catch(() => {
+                snackBarMessage(new Error("Access denied"));
+            })
+    }
+
+    getEc2AuthConfig() {
+        callVaultApi('get', `${this.state.baseVaultPath}/config/client`, null, null)
+            .then((resp) => {
+                let config = resp.data.data;
+                this.setState({
+                    configObj: update(this.state.configObj,
+                        {
+                            access_key: { $set: (config.access_key ? config.access_key : null) },
+                            endpoint: { $set: config.endpoint },
+                            secret_key: { $set: (config.secret_key ? config.secret_key : null) }
+                        }),
+                    newConfigObj: update(this.state.configObj,
+                        {
+                            access_key: { $set: (config.access_key ? config.access_key : null) },
+                            endpoint: { $set: config.endpoint },
+                            secret_key: { $set: (config.secret_key ? config.secret_key : null) }
+                        })
+                });
+            })
+            .catch(snackBarMessage);
```

Comment:
Again, vault returns 404 if backend has never been configured. Worth handling here and displaying a better message like `This backend is not yet configured`